Fix for non-existing revision in many-to-many association

valkars opened the pull request:
## Subject

I am targeting this branch, because this patch is backwards compatible.

I found a bug when viewing the history of an entity that has ManyToMany associations. If a revision for the related entity cannot be found, a NoRevisionFoundException is thrown. In ToOne part of the code `if (self::isToOne($assoc)) {` - this exception is caught and the object is then loaded directly from the database. I added the same handling to the ManyToMany block, and now everything works correctly — in my project, all revisions are available for entities with ManyToMany relations.

## Changelog
```markdown
### Fixed
- NoRevisionFoundException when viewing history of entities with ManyToMany associations.
```

Commit:
Fix for non-existing revision in many-to-many association

valkars commented:
@VincentLanglet when you have some time, could you please review my pull request?

VincentLanglet reviewed (dismissed):
Are you able to write a test for this bug @valkars ?

valkars commented:
I'll try

valkars commented:
Test added @VincentLanglet

VincentLanglet commented:
Just the linter to fix.

valkars commented:
Unfortunately it's not possible, I tried, there is problem with order of phpdoc params, first run on my computer:
```
Using cache file ".php-cs-fixer.cache".
 138/138 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

   1) src/Collection/AuditedCollection.php (phpdoc_order, phpdoc_separation, phpdoc_trim)
   2) src/ChangedEntity.php (phpdoc_order)

Fixed 2 of 138 files in 3.912 seconds, 20.00 MB memory used
```
Then run again and it change these 2 lines backwards:
```
Using cache file ".php-cs-fixer.cache".
 138/138 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

   1) src/Collection/AuditedCollection.php (phpdoc_order, phpdoc_separation, phpdoc_trim)
   2) src/ChangedEntity.php (phpdoc_order)

Fixed 2 of 138 files in 3.902 seconds, 20.00 MB memory used
```
So I have no idea how to fix it :( 
@VincentLanglet

valkars commented:
I have commited with these changes and then fixer again wrote that there are 2 errors. So I reversed these files

VincentLanglet commented:
Ill take a look in the afternoon

VincentLanglet commented on src/Collection/AuditedCollection.php:
```diff
@@ -323,9 +323,10 @@ public function forAll(\Closure $p)
+     * @phpstan-template U
+     *
      * @return Collection<TKey, U>
      *
```
```suggestion
     * @return Collection<TKey, U>
     *
     * @phpstan-template U
```

VincentLanglet committed:
Update src/Collection/AuditedCollection.php

VincentLanglet commented:
I'll wait for https://github.com/sonata-project/EntityAuditBundle/pull/652 before doing a release

valkars commented:
Ok, I'll try to finish it next week
